extensions/mcp/src/getProcessedClients.ts

```diff
@@ -1,49 +1,83 @@
-import { getClients } from "./getClients";
-import os from "os";
-import { execSync } from "child_process";
-import { environment } from "@raycast/api";
 import { Client } from "@modelcontextprotocol/sdk/client/index.js";
-import { StdioClientTransport } from "@modelcontextprotocol/sdk/client/stdio.js";
+import { StdioClientTransport, getDefaultEnvironment } from "@modelcontextprotocol/sdk/client/stdio.js";
+import { environment } from "@raycast/api";
+import { execSync } from "child_process";
+import os from "os";
+import { getClients } from "./getClients";
 
-export default async function (clientName?: string) {
-  function initEnv() {
-    const shell = os.userInfo().shell || "/bin/sh";
-    const command = `LC_ALL=en_US.UTF-8 ${shell} -L -i -c 'printenv'`;
-    try {
-      const variables = execSync(command, { encoding: "utf8" });
-      variables.split("\n").forEach((line) => {
-        const [key, value] = line.split("=");
-        if (key && value) {
-          process.env[key] = value;
-        }
-      });
-    } catch (error) {
-      console.error("Error retrieving shell PATH:", error);
-      process.env.PATH = process.env.PATH || "";
-    }
-  }
-  initEnv();
-
-  function getUserShellPath() {
-    const shell = os.userInfo().shell || "/bin/sh";
-    const command = `${shell} -l -c 'echo $PATH'`;
-
-    try {
-      const path = execSync(command).toString().trim();
-      return path;
-    } catch (error) {
-      console.error("Error retrieving shell PATH:", error);
-      return process.env.PATH || "";
-    }
+/**
+ * Establishes a connection to a MCP Server using StdioClientTransport.
+ * Also redirects the server's stderr to the console.
+ *
+ * @param raycastAiClientInstance - The client instance to connect to
+ * @param transport - The transport to use for the connection
+ */
+async function connectClient(raycastAiClientInstance: Client, transport: StdioClientTransport) {
+  console.log("Connecting to client...");
+  const connectPromise = raycastAiClientInstance.connect(transport);
+  transport?.stderr?.on("data", (data) => {
+    console.log(`MCP Server logs: ${data.toString()}`);
+  });
+
+  await connectPromise;
+  console.log("Connected to client successfully");
+}
+
+/**
+ * Build an environment object that will be used to spawn the MCP server process.
+ * Combines environment variables from the shell and adds critical paths.
+ *
+ * @returns The environment object with properly configured PATH and other variables
+ */
+function buildProcessEnv() {
+  console.log("=== Building process environment ===");
+
+  // Get user's shell
+  const shell = os.userInfo().shell || "/bin/sh";
+  console.debug(`User shell: ${shell}`);
+
+  // Get environment variables from shell
+  try {
+    // Get PATH from shell
+    const pathCommand = `${shell} -l -i -c 'echo $PATH'`;
```

Comment:
can we also remove interactive `-i` flag ?
the command execution never gets completed in my shell with this flag

Reply:
also in src/manage-mcp-servers.tsx#98

Reply:
Okay, removed `-i` flags from both